app/validators/subnets_consistency_validator.rb

```diff
@@ -0,0 +1,15 @@
+class SubnetsConsistencyValidator < ActiveModel::Validator
+  def validate(record)
+    if record.subnet && record.subnet6
```

Comment:
Can be written with a guard clause, e.g.
`return unless record.subnet && record.subnet6`